### PR TITLE
enhance(curriculum): add inline scrim design to custom element

### PR DESCRIPTION
**Pull request by LeoMcA**
## Summary

https://mozilla-hub.atlassian.net/browse/MP-1508

Merge before https://github.com/mdn/curriculum/pull/73

On staging, e.g. https://developer.allizom.org/en-US/curriculum/getting-started/soft-skills/#1.5_succeeding_in_job_interviews

### Problem

Our inline scrims should have custom cover images

### Solution

Create these inline, to save bandwidth (~600kB image per scrim -> ~60kB of assets for *all* scrims)

---

### After

Screen Shot 2024-09-26 at 15 39 01

---

## How did you test this change?

- `yarn dev`
- localhost:3000
- across Firefox, Chrome and Epiphany
- will get on stage to test more

### CODE REVIEW

File: .github/workflows/stage-build.yml [hunk 1]
```diff
@@ -129,6 +129,7 @@ jobs:
         with:
           repository: mdn/curriculum
           path: mdn/curriculum
+          ref: "next"
```
Comment (caugner) [inline .github/workflows/stage-build.yml:132]:
```suggestion
```

File: client/src/curriculum/index.scss [hunk 2]
```diff
@@ -289,9 +289,15 @@
       }
 
       scrim-inline {
+        aspect-ratio: 1.5;
         display: block;
-        height: 14.25rem;
-        width: 22rem;
+        margin: 0.5rem auto;
+        max-width: 575px;
```
Comment (caugner) [inline client/src/curriculum/index.scss:295]: Curious why you chose `px` over `rem` here?

**Review by caugner — APPROVED** [review]
LGTM.

File: client/src/curriculum/index.scss [hunk 3]
```diff
@@ -289,9 +289,15 @@
...
         display: block;
-        height: 14.25rem;
-        width: 22rem;
+        margin: 0.5rem auto;
+        max-width: 36rem;
+        width: 100%;
+      }
+
+      p:has(> scrim-inline:only-child) {
+        margin: 1rem 0;
```
Comment (fiji-flo) [inline client/src/curriculum/index.scss:300]: Do we really need this. Starting to use `:has` would be a the first time here.

File: client/tsconfig.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
     "isolatedModules": true,
     "jsx": "react-jsx",
     "lib": ["dom", "dom.iterable", "es2021"],
-    "module": "ES2020",
+    "module": "ESNext",
```
[the remark this answers is not in the document]
Comment (LeoMcA) [reply client/tsconfig.json:10]: Import attributes require it

File: client/src/curriculum/index.scss (hunk 3 above)
Comment (LeoMcA) [reply client/src/curriculum/index.scss:300]: We don't *need* it, but it's a small visual improvement, which is exactly what `:has` is perfect for.

Without:

image

With:

image

**Pull request merged**